File: manual/data/python/tests/test_jsonschema_iter_errors.py
import unittest
from function_jsonschema_iter_errors import JSONValidator

class TestJSONValidator(unittest.TestCase):
    
    def setUp(self):
        self.validator = JSONValidator()

    def test_case_1(self):
        data = '{"name": "John", "age": 30}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "JSON is valid")

    def test_case_2(self):
        data = '{"name": "John"}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "Validation failed: 'age' is a required property")

    def test_case_3(self):
        data = '{"name": "John", "age": "30"}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "Validation failed: '30' is not of type 'integer'")

    def test_case_4(self):
        data = '{}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "Validation failed: 'name' is a required property")

    def test_case_5(self):
        data = '{"name": "John", "age": 30}'
        schema = '{"type": "object", "properties": {"name": {"type": "string"}}, "required": ["name"]}'
        self.assertEqual(self.validator.validate_json(data, schema), "JSON is valid")

if __name__ == '__main__':
    unittest.main()

File: manual/data/cpp/tests/test_jsonschema_iter_errors.cpp
#include <gtest/gtest.h>
#include "../src/function_jsonschema_iter_errors.cpp"

class TestJSONValidator : public ::testing::Test {
protected:
    JSONValidator validator;
};

TEST_F(TestJSONValidator, TestCase1) {
    std::string data = R"({"name": "John", "age": 30})";
    std::string schema = R"({"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]})";
    EXPECT_EQ(validator.validate_json(data, schema), "JSON is valid");
}

TEST_F(TestJSONValidator, TestCase2) {
    std::string data = R"({"name": "John"})";
    std::string schema = R"({"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]})";
    EXPECT_EQ(validator.validate_json(data, schema), "Validation failed: 'age' is a required property");
}

TEST_F(TestJSONValidator, TestCase3) {
    std::string data = R"({"name": "John", "age": "30"})";
    std::string schema = R"({"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]})";
    EXPECT_EQ(validator.validate_json(data, schema), "Validation failed: '30' is not of type 'integer'");
}

TEST_F(TestJSONValidator, TestCase4) {
    std::string data = R"({})";
    std::string schema = R"({"type": "object", "properties": {"name": {"type": "string"}, "age": {"type": "integer"}}, "required": ["name", "age"]})";
    EXPECT_EQ(validator.validate_json(data, schema), "Validation failed: 'name' is a required property");
}

TEST_F(TestJSONValidator, TestCase5) {
    std::string data = R"({"name": "John", "age": 30})";
    std::string schema = R"({"type": "object", "properties": {"name": {"type": "string"}}, "required": ["name"]})";
    EXPECT_EQ(validator.validate_json(data, schema), "JSON is valid");
}

int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}